### PR TITLE
Improve the documentation about building in the Build Service

### PR DESCRIPTION
**Pull request by dcermak**
Changes proposed in this pull request:
* Incorporate @mlschroe's comments about building in OBS into the documentation

### CODE REVIEW

File: doc/source/building/build_in_buildservice.rst [hunk 1]
```diff
@@ -167,31 +173,57 @@ your dependent packages. These repositories can be provided in two ways:
...
+
+   You will be unable to build images for different distributions from the
+   same OBS project when adding repositories to your project's
+   configuration. The problem is, that all your image builds share the same
+   repositories. This will result in dependency conflicts for different
+   distributions.
+
+   On the other hand this approach requires a lot less workarounds in the
+   project configuration as compared to adding the repositories via the
+   :file:`config.xml`.
```
[the remark this answers is not in the document]
Comment (dcermak) [reply doc/source/building/build_in_buildservice.rst]: I've formulated it differently but tried to follow the ANSI norm. The "issue" with your formulation was imho that it sounds like you shouldn't add the repositories to the project configuration. But that is actually the opposite of what we want: you **should** put them there, as it makes working with OBS a ton easier. It is just something one needs to be aware of.

[the remark this answers is not in the document]
Comment (tomschr) [reply doc/source/building/build_in_buildservice.rst]: Oh, I see! I just created that example to give you an idea about it. Great that you could apply it to this warning. :+1: Thanks Dan! :-)

File: doc/source/building/build_in_buildservice.rst [hunk 2]
```diff
@@ -167,31 +173,60 @@ your dependent packages. These repositories can be provided in two ways:
...
+          <arch>x86_64</arch>
+        </repository>
+      </project>
+
+   Now `Virtualization:Appliances:Builder` is the last repository, which'
+   repositories are included into the search path. As
+   `openSUSE:Factory/snapshot` is among these, it can be omitted from the
+   repository list.
+
+#. Keep the repositories in your :file:`config.xml` configuration file. If
```
Comment (davidcassany) [inline doc/source/building/build_in_buildservice.rst]: Probably i would add  note here with a link to some examples https://build.opensuse.org/project/show/Virtualization:Appliances:Images:Testing_x86 I found them explanatory and there you can also see the _painful_ prjconf configuration.

Comment (dcermak) [reply doc/source/building/build_in_buildservice.rst]: That repository is already mentioned at the bottom of the same file in the section Project configuration. I could add a link to that section?

Comment (davidcassany) [reply doc/source/building/build_in_buildservice.rst]: mmm I think this is not the same repository. Note that this is the repository for KIWI testing images, where we do not use `obsrepositories:/` trick so we are capable to build several distro in the same project at a cost of a hard to determine prjconf configuration.

Comment (dcermak) [reply doc/source/building/build_in_buildservice.rst]: Ah, sorry, I was thinking you are referring to a different section. 

So the thing is, I am actually recommending here (as per @mlschroe's recommendation) to keep your normal repositories in `config.xml` and to put the kiwi repository into OBS' config. That will make it impossible to build multiple distros from the same project, too, but it should make working with OBS a lot easier (and it shouldn't require all these nasty `Prefer:`/`Substitute:`/etc. in the project config). Then referring to https://build.opensuse.org/project/show/Virtualization:Appliances:Images:Testing_x86 wouldn't be too helpful, as there these workarounds are unfortunately necessary, as the kiwi repo is in `config.xml` and not in the obs repository configuration.

**Pull request merged**